src/tribler/core/components/database/database_component.py

```diff
@@ -1,23 +1,67 @@
+from tribler.core import notifications
 from tribler.core.components.component import Component
+from tribler.core.components.database.db.store import MetadataStore
 from tribler.core.components.database.db.tribler_database import TriblerDatabase
+from tribler.core.components.key.key_component import KeyComponent
+from tribler.core.components.knowledge.rules.knowledge_rules_processor import KnowledgeRulesProcessor
 from tribler.core.utilities.simpledefs import STATEDIR_DB_DIR
 
 
 class DatabaseComponent(Component):
     tribler_should_stop_on_component_error = True
 
     db: TriblerDatabase = None
+    mds: MetadataStore = None  # TODO: legacy, should be merged into ``db``
 
     async def run(self):
         await super().run()
 
-        db_path = self.session.config.state_dir / STATEDIR_DB_DIR / "tribler.db"
-        if self.session.config.gui_test_mode:
+        config = self.session.config
+        db_path = config.state_dir / STATEDIR_DB_DIR / "tribler.db"
+        if config.gui_test_mode:
             db_path = ":memory:"
 
         self.db = TriblerDatabase(str(db_path))
 
+        # TODO: merge the code below into the TriblerDatabase
```

Comment:
Is this a TODO for the next PR, or is it a forgotten TODO for the current PR?

Reply:
This is for the next PR.

Reply:
Some of the developers and I in this group consider it bad, or let's say "not good," practice to merge changes with TODOs into the main branch.

I'm not stopping you from doing this by not giving acceptance to the PR. I'm just highlighting that there are many points of view on this practice, some of which consider it to be a bad practice that leads to code quality degradation.

https://www.reddit.com/r/learnprogramming/comments/r25bj7/are_is_committing_code_to_github_with_todos_bad/?rdt=46532

Reply:
I agree with them. However, our current linters are configured to allow this. Notably, I did not have to disable any warnings to pass our PR checks.

Once we fix https://github.com/Tribler/tribler/issues/1722 we can adopt a more strict stance on this.

Reply:
Are you saying that if you understand there is a bad practice, but there are no configured linter issues, then it is okay for you to commit and merge the code despite its quality?

Reply:
Even though I disagree with the practice personally, this is the approach that the team has chosen and I adapt my workflow to it.

If I were the only developer, I would make make own standards and conform to what I would like to see. Sadly, I am not the only developer and I have to conform to the existing rules and linter settings.